src/init.cpp

```diff
@@ -1276,6 +1276,68 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
                                      chainman, *node.mempool, ignores_incoming_txs);
     RegisterValidationInterface(node.peerman.get());
 
+    // Check port numbers
+    if (args.IsArgSet("-port")) {
+        uint64_t portIn = args.GetIntArg("-port", 0);
+        if (portIn > std::numeric_limits<uint16_t>::max())
+            return InitError(strprintf(_("Invalid port specified in -port: '%d'"), portIn));
```

Comment:
This should also error for port 0.

See discussions in https://github.com/bitcoin/bitcoin/pull/24116 and https://github.com/bitcoin/bitcoin/pull/24344

Reply:
And for less than 0... (change type above to `int64_t`)

Reply:
Thanks for the input, guys. I applied your suggestions + set `bool SplitHostPort(...)` to consider port 0 as invalid.